tools/clang/tools/dxil2spv/dxil2spvmain.cpp

```diff
@@ -0,0 +1,89 @@
+//===------- dxil2spv.cpp - DXIL to SPIR-V Tool -----------------*- C++ -*-===//
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//===----------------------------------------------------------------------===//
+//
+//  This file provides the entry point for the dxil2spv executable program.
+//
+//===----------------------------------------------------------------------===//
+
+#include "dxc/DXIL/DxilUtil.h"
+#include "dxc/DxilContainer/DxilContainer.h"
+#include "dxc/DxilContainer/DxilContainerReader.h"
+#include "dxc/Support/ErrorCodes.h"
+#include "dxc/Support/Global.h"
+#include "dxc/Support/HLSLOptions.h"
+#include "dxc/Support/dxcapi.use.h"
+#include "dxc/dxcapi.h"
+
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IRReader/IRReader.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/raw_ostream.h"
+
+static dxc::DxcDllSupport dxcSupport;
+
+#ifdef _WIN32
+int __cdecl wmain(int argc, const wchar_t **argv_) {
+#else
+int main(int argc, const char **argv_) {
+#endif // _WIN32
+  if (argc < 2) {
+    fprintf(stderr, "Required input file argument is missing.\n");
+    return DXC_E_GENERAL_INTERNAL_ERROR;
+  }
+
+  hlsl::options::StringRefUtf16 filename(argv_[1]);
```

Comment:
In high level, if you added a document (even though it is incomplete and simple), it would be helpful.
Maybe `docs/DXIL-to-SPIRV.rst` ??

If you think it is not ready (or not needed at this time), please briefly add a description as a comment in this file that explains
- Input e.g., DXIL bitcode file??
- Output e.g., SPIR-V file?? plain text file with SPIR-V code??
- or say "TODO" if it is not supported yet
- How to build / run e.g., a briefly help message

Reply:
I started writing a wiki and rst doc, but I think it just doesn't make sense until there's actual output (this is just the very beginning piece, trying to keep PRs a manageable size). Documented build/run in a comment for now like you suggested.